rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/interpreter/DBFileMap.kt

```diff
@@ -64,6 +64,9 @@ class DBFileMap {
  * */
 data class EnrichedDBFile(private val dbFile: DBFile, private val fileDefinition: FileDefinition, val jarikoMetadata: FileMetadata) : DBFile {
 
+    // All files are opened by default when defined in F specs.
+    var open = true
```

Comment:
My doubt is about the necessity or less of a full implementation of OPEN/CLOSE statement. In actual implementation, OPEN and CLOSE modify only the state of "open" variable into EnrichDBFile. There are not controls about the effect of this state modification and files are always considerated opened, also when there was a CLOSE statement call that set open variable to false. Is it necessary a fully implementation of OPEN/CLOSE?

Reply:
One thing is full, for example opening the file (open connection) on open, and closing the file on close, another think is handle the `DBEnrichedFile.state`.
I agree about what you say, but since that you have introduced a little bit of complexity, if you don't manage  `DBEnrichedFile.state` that complexity is noise.
For instance a simple and cheap way to give a sense to `state` could be implementing inside `DBEnrichedFile` this function:

```kotlin
    private fun DBFile.verifyConnection(): DBFile {
        if (!open) {
            throw IllegalStateException("File $name is not open")
        }
        return this
    }
```

and changing all method in `DBEnrichedFile` that need connection open to work property like this:
```kotlin
    private fun DBFile.verifyConnection(): DBFile {
        if (!open) {
            throw IllegalStateException("File $name is not open")
        }
        return this
    }
```

Reply:
Let me know what do you think

Reply:
already had implemented a solution similar to yours, with this new function defined in class EnrichedDbFile

```kotlin
private fun checkOpened(): DBFile {
        require(open) {
            "Cannot access to closed file $name"
        }
        return dbFile
    }
```

and DBFile methods override calls like this

```kotlin
override fun write(record: Record) = checkOpened().write(record).validate()
```

I'm not a fan of extension function...